pkg/hubble/relay/server/option.go

```diff
@@ -212,3 +212,13 @@ func WithClientMTLSFromCert(ca *x509.CertPool, cert tls.Certificate) Option {
 		MinVersion:   ClientTLSMinVersion,
 	})
 }
+
+// WithInsecureClient disables transport security for connection to Hubble
+// server instances. Transport security is required to WithInsecureClient is
+// set (not recommended).
+func WithInsecureClient() Option {
```

Comment:
nit: I find the client/server terminology a bit ambiguous here. Yes, hubble-relay acts as a client, but it also has clients such as UI/CLI that connect to it.

Can we call the client a `peer` instead, i.e. name this `WithInsecurePeer`?

Reply:
That's a valid point. I'll try to come up with a terminology that is consistent and not confusing and will amend the PR.

Reply:
@kAworu opened #13094